x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -542,17 +619,24 @@ export class ESSearchSource extends AbstractESSource {
     };
   }
 
-  getSyncMeta() {
+  getSyncMeta(): VectorSourceSyncMeta | null {
     return {
-      sortField: this._descriptor.sortField,
-      sortOrder: this._descriptor.sortOrder,
+      sortField: typeof this._descriptor.sortField === 'string' ? this._descriptor.sortField : '',
+      sortOrder: this._descriptor.sortOrder ? this._descriptor.sortOrder : SORT_ORDER.ASC,
       scalingType: this._descriptor.scalingType,
-      topHitsSplitField: this._descriptor.topHitsSplitField,
-      topHitsSize: this._descriptor.topHitsSize,
+      topHitsSplitField:
+        typeof this._descriptor.topHitsSplitField === 'string'
+          ? typeof this._descriptor.topHitsSplitField
+          : '',
+      topHitsSize:
```

Comment:
Logic is not needed, `descriptor.topHitsSize` is provided a default value in the constructor